scipy/spatial/transform/_rigid_transform_xp.py

```diff
@@ -245,6 +247,30 @@ def pow(matrix: Array, n: float | Array) -> Array:
     return from_exp_coords(as_exp_coords(matrix) * n)
 
 
+def mean(matrix: Array, weights: ArrayLike | None = None) -> Array:
+    xp = array_namespace(matrix)
+    if matrix.shape[0] == 0:
+        raise ValueError("Mean of an empty rotation set is undefined.")
+
+    quats = quat_from_matrix(matrix[..., :3, :3])
```

Comment:
More food for thoughts on optimizations we could do: At the moment, every call to `from_matrix` is checking if the matrix is a valid rotation matrix, and normalizes the matrix for lazy frameworks. In cases like this where we call the function from within our own backend, the matrix is guaranteed to be valid and there is no need to normalize again. We could have a `from_matrix_unsafe` function that implements only the conversion from a valid matrix to a quaternion, and then only normalize inputs from outside the backend.

Not relevant for this PR, but it could significantly improve performance.

Reply:
Agreed, I like this idea quite a bit